src/ott/problems/linear/potentials.py

```diff
@@ -99,11 +102,13 @@ def transport(self, vec: jnp.ndarray, forward: bool = True) -> jnp.ndarray:
     from ott.geometry import costs
 
     vec = jnp.atleast_2d(vec)
+
     if self._corr and isinstance(self.cost_fn, costs.SqEuclidean):
       return self._grad_f(vec) if forward else self._grad_g(vec)
+    twist_op = jax.vmap(self.cost_fn.twist_operator, in_axes=[0, 0, None])
     if forward:
-      return vec - self._grad_h_inv(self._grad_f(vec))
-    return vec - self._grad_h_inv(self._grad_g(vec))
+      return twist_op(vec, self._grad_f(vec), 0)
```

Comment:
because I think 0 and 1 are used here for either of the two possible variable choices (rather than ``False`` and ``True``).

Could be passed as `int` if more clear.

Reply:
Don't think passing as `int` is more clear, as only 2 values can be passed, would stick with bool.